### PR TITLE
Http proxy bypass

### PR DESCRIPTION
**Pull request by willmeek**
This PR builds on https://github.com/puppetlabs/puppetlabs-apt/pull/699

Rather than always defaulting the https proxy to 'DIRECT', use another proxy setting to determine whether this should be the case.

### CODE REVIEW

File: spec/classes/apt_spec.rb [hunk 1]
```diff
@@ -84,7 +84,7 @@
         is_expected.to contain_apt__setting('conf-proxy').with(priority: '01').with_content(
           /Acquire::http::proxy "http:\/\/localhost:8180\/";/,
         ).without_content(
-          %r{Acquire::https::proxy},
+          /Acquire::https::proxy/,
```
Comment (tphoney) [inline spec/classes/apt_spec.rb]: this should still be the %r{ }

File: spec/classes/apt_spec.rb [hunk 2]
```diff
@@ -72,7 +72,7 @@
         is_expected.to contain_apt__setting('conf-proxy').with(priority: '01').with_content(
           /Acquire::http::proxy "http:\/\/localhost:8080\/";/,
         ).without_content(
-          %r{Acquire::https::proxy},
+          /Acquire::https::proxy/,
```
Comment (tphoney) [inline spec/classes/apt_spec.rb]: this should still be the %r{ }

**Comment by tphoney** [comment]
Otherwise looks good !!

File: spec/classes/apt_spec.rb (hunk 1 above)
Comment (willmeek) [reply spec/classes/apt_spec.rb]: Updated all regexes to use %r notation

File: spec/classes/apt_spec.rb (hunk 2 above)
Comment (willmeek) [reply spec/classes/apt_spec.rb]: Updated all regexes to use %r notation

**Comment by eputnam** [comment]
Would you mind throwing a little blurb about it here? https://github.com/puppetlabs/puppetlabs-apt/blame/master/README.md#L284 I had to use git blame to get a stupid line number.

**Comment by willmeek** [comment]
@eputnam Thanks Eric, great suggestion, updated with the new setting.

**Pull request merged**